src/components/App/ModalApp.tsx

```diff
@@ -174,6 +177,10 @@ class ModalApp extends Component<Props> {
       const validEnterpriseFeatures = getEnterpriseFeaturesFromJWT(token)
       this.setConfiguredEnterpriseFeatures(validEnterpriseFeatures, options)
     }
+
+    if (customUI && customUI !== prevCustomUI) {
```

Comment:
I think you would want to deep compare objects here, so probably something like using https://lodash.com/docs/4.17.15#isEqual or equivalent.

Reply:
First, please no Lodash.
SDK is size sensitive and Lodash weighs a ton.

Second, and most importantly, I'd argue deep compare is a bad idea.
Not only because it is computationally expensive, but mainly because it encourages object mutations by supporting it.
Immutability and reactivity, I say.

Reply:
Lodash was just an example.

But isn't `customUI !== prevCustomUI` always truthy though? React will always emit a new object, isn't it?

Reply:
As long as `prepareInitialStore` is always called with a pointer to the same object in memory, and `prevOptions` doesn't mutate its `customUI` object reference, they should be stable.

Which means, as long as `prepareInitialStore` isn't called with a new `customUI` in its `options`, the objects will always be the same.

Reply:
Fine, ignore me, I'm like on Redux now thinking it emits a new immutable object each time.

Reply:
Even in Redux if the reducer doesn't touch `customUI` it's still the same.
The base store object might change, but the value of one of its properties can still hold a pointer to the same object.

Reply:
For example:

```ts
const customUI = { foo: 'bar' };
const a = { customUI };
let b = { customUI };

a.customUI === b.customUI;
// true

b = { ...b, customUI };

a.customUI === b.customUI;
// true
```